storm-client/src/jvm/org/apache/storm/Config.java

```diff
@@ -304,11 +307,13 @@
     // an error will be thrown by nimbus on topology submission and not by the client prior to submitting
     // the topology.
     public static final String TOPOLOGY_SCHEDULER_STRATEGY = "topology.scheduler.strategy";
+
     /**
-     * Declare scheduling constraints for a topology used by the constraint solver strategy. A List of pairs (also a list) of components
-     * that cannot coexist in the same worker.
+     * Declare scheduling constraints for a topology used by the constraint solver strategy. The format can be either
+     * old style (validated by ListOfListOfStringValidator.class or the newer style, which is a list of specific type of
+     * Maps (validated by RasConstraintsTypeValidator.class). The value must be in one or the other format.
```

Comment:
Some examples of old style and new style configurations might help.

Reply:
Added